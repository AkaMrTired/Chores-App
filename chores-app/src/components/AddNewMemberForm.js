/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from "react";
import emailjs from "@emailjs/browser";
import axios from "axios";

const AddNewMemberForm = () => {
  const familyID = localStorage.getItem("familyID");

  const initialState = {
    fields: {
      email: "",
      name: "",
      role: "Choose a role",
    },
  };
  const [fields, setFields] = useState(initialState.fields);
  const [roleError, setRoleError] = useState();
  const [message, setMessage] = useState();
  const sendEmail = () => {
    const initialLink = `http://localhost:3000/newmembersignup?name=${fields.name}&email=${fields.email}`;
    const noSpacesLink = initialLink.replace(/\s+/g, "");
    const emailParams = {
      link: noSpacesLink,
      email: fields.email,
    };
    axios
      .post(`http://localhost:3300/family/${familyID}/users`, {
        name: fields.name,
        role: fields.role,
        email: fields.email,
      })
      .then(() => {
        return emailjs.send(
          process.env.REACT_APP_EMAIL_SERVICE_ID,
          process.env.REACT_APP_EMAIL_TEMPLATE_ID,
          emailParams,
          process.env.REACT_APP_EMAIL_PUBLIC_KEY
        );
      })
      .then(() => {
        setMessage("Invite sent, add another or return to dashboard");
        setFields(initialState.fields);
      })
      .catch(() => {
        setMessage(`uh oh, it looks like there was an error`);
      });
  };

  const handleSubmit = (event) => {
    if (fields.role !== "Choose a role") {
      event.preventDefault();
      sendEmail();
    } else {
      event.preventDefault();
      setRoleError("Please select a role for the person you are inviting");
    }
  };

  const handleFieldChange = (event) => {
    setMessage();
    event.preventDefault();
    setFields({ ...fields, [event.target.name]: event.target.value });
    setRoleError();
  };

  return (
    <div className="sign-up-container">
      <h1>Invite New Member</h1>
      <div>
        <form onSubmit={handleSubmit}>
          <label htmlFor="name">Name </label>
          <input
            name="name"
            required
            type="text"
            placeholder="e.g Timmy"
            value={fields.name}
            onChange={handleFieldChange}
          />
          <label htmlFor="role">Role</label>
          <select
            required
            name="role"
            value={fields.role}
            onChange={handleFieldChange}
          >
            <option value="Choose a role">Choose a Role</option>

            <option value="parent">Parent</option>
            <option value="child">Child</option>
          </select>
          {!!roleError && <p>{roleError}</p>}
          <label htmlFor="email">Email</label>
          <input
            type="email"
            name="email"
            required
            placeholder="e.g example@example.com"
            value={fields.email}
            onChange={handleFieldChange}
          />
          <button type="submit" className="btn btn-fill_purple">
            Send Invite!
          </button>

          {!!message && <p>{message}</p>}
        </form>
      </div>
      <div className="back-btn-wrapper">
        <button type="button" className="btn btn-stroke_purple">
          <a href="/parentdashboard">Back To Dashboard</a>
        </button>
      </div>
    </div>
  );
};

export default AddNewMemberForm;
