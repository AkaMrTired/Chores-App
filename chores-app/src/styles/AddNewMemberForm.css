.invite-member-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    align-self: center;
    background: var(--gradient);
    box-shadow: var(--shadow);
    padding: 3rem 1rem;
    border-radius: 20px;
    width: 90vw;
    max-width: 1000px;
    margin: 5rem;
}

.invite-form {
    display: flex;
    flex-direction: column
}