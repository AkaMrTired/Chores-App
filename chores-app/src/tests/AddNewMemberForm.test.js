import { render, screen } from "@testing-library/react";
import React from "react";
import AddNewMemberForm from "../components/AddNewMemberForm";

describe("the new member form", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<AddNewMemberForm />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(<AddNewMemberForm />);
    expect(screen.getByText(/new member/i)).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /send invite/i })
    ).toBeInTheDocument();
    expect(screen.getByText(/choose a role/i)).toBeInTheDocument();
    expect(screen.getByText(/email/i)).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /back to dashboard/i })
    ).toBeInTheDocument();
  });
});
