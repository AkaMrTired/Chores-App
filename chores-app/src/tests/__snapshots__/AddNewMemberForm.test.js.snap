// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the new member form matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sign-up-container"
  >
    <h1>
      Invite New Member
    </h1>
    <div>
      <form>
        <label
          for="role"
        >
          Role
        </label>
        <select
          name="role"
          required=""
        >
          <option
            value="Choose a role"
          >
            Choose a Role
          </option>
          <option
            value="parent"
          >
            Parent
          </option>
          <option
            value="child"
          >
            Child
          </option>
        </select>
        <label
          for="email"
        >
          Email
        </label>
        <input
          name="email"
          placeholder="e.g example@example.com"
          required=""
          type="email"
          value=""
        />
        <button
          type="submit"
        >
          Send Invite!
        </button>
      </form>
    </div>
    <div>
      <button
        type="button"
      >
        <a
          href="/parentdashboard"
        >
          Back To Dashboard
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
